tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1623,33 +1640,45 @@ void ROOT::Experimental::Internal::RNTupleFileWriter::WriteBareFileSkeleton(int
+   // buffer will point to the wrong places.
+
+   // Figure out key length
    RTFString strTList{"TList"};
    RTFString strStreamerInfo{"StreamerInfo"};
    RTFString strStreamerTitle{"Doubly linked list"};
-
    fFileSimple.fControlBlock->fHeader.SetSeekInfo(fFileSimple.fKeyOffset);
-   RTFKey keyStreamerInfo(fFileSimple.fControlBlock->fHeader.GetSeekInfo(), 100, strTList, strStreamerInfo,
-                          strStreamerTitle, 0);
-   RTFStreamerInfoList streamerInfo;
-   auto classTagOffset = keyStreamerInfo.fKeyLen + offsetof(struct RTFStreamerInfoList, fStreamerInfo) +
-                         offsetof(struct RTFStreamerInfoObject, fStreamers) +
-                         offsetof(struct RTFStreamerVersionEpoch, fNewClassTag) + 2;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerVersionMajor.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerVersionMinor.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerVersionPatch.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekHeader.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesHeader.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenHeader.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekFooter.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesFooter.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenFooter.fClassTag = 0x80000000 | classTagOffset;
-   streamerInfo.fStreamerInfo.fStreamers.fStreamerMaxKeySize.fClassTag = 0x80000000 | classTagOffset;
+   auto keyLen =
+      RTFKey(fFileSimple.fControlBlock->fHeader.GetSeekInfo(), 100, strTList, strStreamerInfo, strStreamerTitle, 0)
+         .fKeyLen;
+
+   TBufferFile buffer(TBuffer::kWrite, keyLen + 1);
+   buffer.SetBufferOffset(keyLen);
+   streamerInfoList.Streamer(buffer);
+   assert(buffer.Length() > keyLen);
+   const auto bufPayload = buffer.Buffer() + keyLen;
+   const auto lenPayload = buffer.Length() - keyLen;
+
    RNTupleCompressor compressor;
-   auto szStreamerInfo = compressor.Zip(&streamerInfo, streamerInfo.GetSize(), 1);
-   fFileSimple.WriteKey(compressor.GetZipBuffer(), szStreamerInfo, streamerInfo.GetSize(),
+   auto zipStreamerInfos = std::make_unique<unsigned char[]>(lenPayload);
+   auto szZipStreamerInfos = compressor.Zip(bufPayload, lenPayload, 1, zipStreamerInfos.get());
+
+   fFileSimple.WriteKey(zipStreamerInfos.get(), szZipStreamerInfos, lenPayload,
                         fFileSimple.fControlBlock->fHeader.GetSeekInfo(), 100, "TList", "StreamerInfo",
                         "Doubly linked list");
-   fFileSimple.fControlBlock->fHeader.SetNbytesInfo(fFileSimple.fFilePos -
+   fFileSimple.fControlBlock->fHeader.SetNbytesInfo(fFileSimple.fKeyOffset -
```

Comment:
just curious: These two should be identical, right? Or was this a sleeping bug that is fixed here?

Reply:
Yes, should be the same. Reverted it back.